src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -137,8 +187,11 @@ const useFetchChannelHook = () => {
     }
 
     try {
-      await saveAllChannelData(signedChannel ?? [], 'SIGNED');
+      if (Array.isArray(signedChannel) && signedChannel.length === 0) return;
+      await saveAllChannelData(signedChannel, 'SIGNED');
       refresh('channelList');
+      const timestamp = new Date().toISOString().split('T')[0];
+      StorageUtils.channelSignedTimeStamps.set(timestamp);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of a timestamp to track when the channels were last fetched is a good practice for caching data. However, it would be better if there was some error handling around the `StorageUtils.channelSignedTimeStamps.set(timestamp);` line in case there are any issues with setting the timestamp in storage.



<!-- This is an auto-generated comment by OSS CodeRabbit -->